src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/HostDiagnosticAnalyzerProvider.cs

```diff
@@ -3,15 +3,64 @@
 // See the LICENSE file in the project root for more information.
 
 using System.Collections.Immutable;
+using System.Reflection;
 using Microsoft.CodeAnalysis.Diagnostics;
 using Microsoft.CodeAnalysis.Workspaces.ProjectSystem;
+using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.LanguageServer.HostWorkspace;
 internal class HostDiagnosticAnalyzerProvider : IHostDiagnosticAnalyzerProvider
 {
+    /// <summary>
+    /// The full path to the Razor source generator
+    /// </summary>
+    public string? RazorSourceGenerator { get; set; }
+
+    private ImmutableArray<(AnalyzerFileReference reference, string extensionId)>? _cachedAnalyzerReferences;
+
+    public HostDiagnosticAnalyzerProvider(string? razorSourceGenerator)
+    {
+        RazorSourceGenerator = razorSourceGenerator;
+    }
+
     public ImmutableArray<(AnalyzerFileReference reference, string extensionId)> GetAnalyzerReferencesInExtensions()
     {
-        // Right now we don't expose any way for the extensions in VS Code to provide analyzer references.
-        return ImmutableArray<(AnalyzerFileReference reference, string extensionId)>.Empty;
+        if (_cachedAnalyzerReferences != null)
+        {
+            return _cachedAnalyzerReferences.Value;
+        }
+
+        if (RazorSourceGenerator == null)
+        {
+            _cachedAnalyzerReferences = ImmutableArray<(AnalyzerFileReference reference, string extensionId)>.Empty;
+            return _cachedAnalyzerReferences.Value;
+        }
+
+        var analyzerReferences = new List<(AnalyzerFileReference, string)>();
+
+        // Create an AnalyzerFileReference for each file
+        var analyzerReference = new AnalyzerFileReference(RazorSourceGenerator, new SimpleAnalyzerAssemblyLoader());
```

Comment:
to confirm, there's only 1 source generator now (the merged version)?  And the project systemproject stuff is responsible for mapping the SDK generator to this one?

Reply:
yes I believe so.